service/security/shiro/src/main/java/org/eclipse/kapua/service/authentication/shiro/utils/AuthenticationUtils.java

```diff
@@ -77,60 +88,31 @@ public static String cryptCredential(CryptAlgorithm algorithm, String plainValue
         return cryptedValue;
     }
 
-    private static String doSha(String plainValue) {
-        try {
-            //
-            // Retrieve Crypt Settings
-            KapuaCryptoSetting settings = KapuaCryptoSetting.getInstance();
-            int saltLength = settings.getInt(KapuaCryptoSettingKeys.CRYPTO_SHA_SALT_LENGTH);
-            String shaAlgorithm = settings.getString(KapuaCryptoSettingKeys.CRYPTO_SHA_ALGORITHM);
-
-            //
-            // Generate salt
-            SecureRandom random = SecureRandom.getInstance("SHA1PRNG");
-            byte[] bSalt = new byte[saltLength];
-            random.nextBytes(bSalt);
-            String salt = Base64.encodeToString(bSalt);
-
-            //
-            // Hash value
-            String hashedValue;
-            switch (shaAlgorithm) {
-                case "SHA256":
-                    hashedValue = (new Sha256Hash(plainValue, salt)).toHex();
-                    break;
-                case "SHA512":
-                default:
-                    hashedValue = (new Sha512Hash(plainValue, salt)).toHex();
-                    break;
-            }
-
-            //
-            // Return value
-            return salt + ":" + hashedValue;
-        } catch (NoSuchAlgorithmException e) {
-            throw new KapuaRuntimeException(KapuaAuthenticationErrorCodes.CREDENTIAL_CRYPT_ERROR, e);
+    public static String doSha(String plainValue) {
+        KapuaCryptoSetting settings = KapuaCryptoSetting.getInstance();
```

Comment:
Please avoid instantiating singletons within the method's body. Instantiate them on top of the class - that makes replacing them with proper Injection much easier in the future

Reply:
old code, I just removed SecureRandom instantiation inside the method since it's a costly operation